File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.java
import java.util.Arrays;

class Solution {
  public int partitionArray(int[] nums, int k) {
    if (nums == null || nums.length == 0) return 0;

    Arrays.sort(nums);
    int count = 1;
    int start = nums[0];

    for (int i = 1, n = nums.length; i < n; i++) {
      if (nums[i] - start > k) {
        count++;
        start = nums[i];
      }
    }

    return count;  }
}

File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.py
from typing import List

class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        if not nums:
            return 0

        nums.sort()
        count = 1
        start = nums[0]

        for num in nums[1:]:
            if num - start > k:
                count += 1
                start = num

        return count
